src/org/zaproxy/zap/extension/ascanrulesBeta/HPP.java

```diff
@@ -154,7 +157,12 @@ public void scan() {
 					log.debug("Injecting payload...");
 					HttpMessage newMsg = getNewMsg();
 					newMsg.setGetParams(tags);
-					sendAndReceive(newMsg);
+					try {
+						sendAndReceive(newMsg);
+					} catch (IllegalStateException|UnknownHostException ex) {
+						log.info("Caught " + ex.getClass().getName() + " when accessing: " + newMsg.getRequestHeader().getURI().toString() + 
+								"\n The target may have replied with a poorly formed redirect due to our input.");
```

Comment:
Not in a new log statement, it should be logged in the same message, after the name of the exception?
To give something like:
`Caught "java.net.UnknownHostException: &zap=zaproxy" when accessing...`
or
`Caught "java.lang.IllegalStateException: unsupported protocol: '172.17.0.2'" when accessing...`